kyo-data/shared/src/main/scala/kyo/Chunk.scala

```diff
@@ -180,7 +182,7 @@ sealed abstract class Chunk[A] extends Seq[A] derives CanEqual:
       * @return
       *   an Iterator[A] over the elements of the Chunk
       */
-    def iterator: Iterator[A] = toArray.iterator
+    def iterator: Iterator[A] = toIndexed.iterator
```

Comment:
This method was quite expensive, always creating a new array just to provide the iterator